components/brave_wallet/browser/eth_logs_tracker.cc

```diff
@@ -50,7 +50,8 @@ void EthLogsTracker::RemoveObserver(EthLogsTracker::Observer* observer) {
 }
 
 void EthLogsTracker::GetLogs() {
-  const auto chain_id = json_rpc_service_->GetChainId(mojom::CoinType::ETH);
+  const auto chain_id =
+      json_rpc_service_->GetChainId(mojom::CoinType::ETH, absl::nullopt);
```

Comment:
This is renderer initiated tracker so should have origin captured upon subscription. And then also should when chain id is changed active subscriptions should stop.
Alternatively chain_id should be captured at the moment subscription starts.

Reply:
fixed 5bea3ae94d98de046b0eb32c9074de4fefcd173d